src/garage/torch/algos/vpg.py

```diff
@@ -261,23 +307,29 @@ def _compute_policy_entropy(self, obs):
 
         return policy_entropy
 
-    def _compute_objective(self, advantages, valids, obs, actions, rewards):
+    def _compute_objective(self, advantages, obs, actions, rewards):
         """Compute objective value.
 
         Args:
-            advantages (torch.Tensor): Expected rewards over the actions.
-            valids (list[int]): Array of length of the valid values.
+            advantages (torch.Tensor): Advantage value at each step.
             obs (torch.Tensor): Observation from the environment.
-            actions (torch.Tensor): Predicted action.
-            rewards (torch.Tensor): Feedback from the environment.
+            actions (torch.Tensor): Actions fed to the environment.
+            rewards (torch.Tensor): Acquired rewards.
 
         Returns:
             torch.Tensor: Calculated objective values
 
         """
-        # pylint: disable=unused-argument
+        del rewards
         log_likelihoods = self.policy.log_likelihood(obs, actions)
-        return log_likelihoods * advantages
+
+        objectives = log_likelihoods * advantages
+
+        if self._entropy_regularzied:
```

Comment:
`_entropy_regularzied` is a wrong spelling. Also, is there a strong reason to move regularization in `_compute_objective`? I think this function is originally designed to compute just the surrogate, and all other stuff are left to `compute_loss`

Reply:
Because I think `self._entropy_regularized` is more directly related to objectives than loss.
In the code, the code only change the value of objectives.

(similarly, I move `self._use_softplus_entropy` to `_compute_policy_entropy` and `_center_adv` and `_positive_adv` to `_compute_advantage`)

Reply:
the objective and the loss are basically the same (loss = -objective) so i think there's too much arguing about semantics here.

as for entropy regularization, it would be more consistent with the literature to put this directly at the loss calculation, because the entropy term is a single scalar which is independent of individual-step advantages, i.e. 

`L' = -objective - mean(H(pi))`